agent/src/android/hooking.ts

```diff
@@ -63,6 +73,39 @@ export namespace hooking {
     });
   };
 
+
+  export const getClassMethodsOverloads = (className: string, methodsAllowList: string[] = []): Promise<JSON> => {
+    return wrapJavaPerform(() => {
+      const clazz: JavaClass = Java.use(className);
+      const methods = clazz.class.getDeclaredMethods()
+      const result = {}
+      methods.forEach(method => {
+        // This trims out only the function name and uses that to get the overloads
+        const methodName = (method.toGenericString().split('.').filter(part => part.includes('('))[0].split('(')[0])
+        if (methodsAllowList.length === 0 || (methodsAllowList.length > 0 && methodsAllowList.includes(methodName))) {
+          const overloads = clazz[methodName].overloads
+          result[methodName] = {
```

Comment:
I'll have to pull the PR to test locally, but does it not make sense to just map the full overloads into this object instead of extracting the args & return only? That way we can bubble up other things to the CLI/API if needed.